lib/puppet/agent.rb

```diff
@@ -40,10 +42,12 @@ def run(client_options = {})
       splay client_options.fetch :splay, Puppet[:splay]
       result = run_in_fork(should_fork) do
```

Comment:
We might want to make this like
```rb
 result = run_with_timeout(should_fork) do
```

And then if `should_fork` is false using `Timeout` and if true using `wait_with_pid`

Reply:
Currently, I'm looking at adding the Timeout inside the `lock { }` block where we run the configurer. This would allow one piece of logic to handle both the forking and non-forking cases.